### PR TITLE
Feature/local usage

### LINKED ISSUE #91: Local testing utilities
**Is your feature request related to a problem? Please describe.**
We would like to use the hlf-network from outside the cluster locally. The current workflow is tedious.

**Describe the solution you'd like**
Provide easy workflow for using the hlf-network locally from outside of the cluster.

### PR DESCRIPTION
**Pull request by palkra**
**Reason for this PR:**
 - Closes #91 

**Changes in this PR:**
 - Add functionality to ```-t``` flag that creates a connection profile for local usage of the network
 - Add label to all cluster worker nodes and function to get current worker node internal ip
 - Add output of cluster worker internal IP to end of the output
 - Worker node IP variable renamed to ```UC4_KIND_NODE_IP``` 
 - Update README with instructions

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -55,6 +55,21 @@ The ```-b``` tag can be used to specify a chaincode tag or branch (develop is de
...
 The script offers the same flags as ```deploy.sh```.
 
+### Local Testing from Outside the Cluster
+
+To test your application locally without deploying it to the cluster, deploy the network (and cluster) with the testing flag: ```source ./restart -t``` (you need to source to keep the node ip environment variable).
+This has the following side-effects: 
+ - We generate a connection profile at `assets/connection_profile_kubernetes_local.yaml` that can be used to access the network from outside the cluster on your machine.
+ - We register a test user with the fixed credentials: `test-admin:test-admin-pw`.
+ - We provide all root certificates at `/tmp/hyperledger/`.
+ - You can export the internal worker node ip as `UC4_KIND_UC4_KIND_NODE_IP`. The last line of the output will look something like this:
```
Comment (NikoBergemann) [inline README.md:65]:
```suggestion
 - You can export the internal worker node ip as `UC4_KIND_NODE_IP`. The last line of the output will look something like this:
```

**Review by NikoBergemann — APPROVED** [review]

**Comment by NikoBergemann** [comment]
why are some testing blocks using scripts/startNetwork/setupLocalNodeIP.sh and some are implementing the same thing themselves?

File: deploy.sh [hunk 2]
```diff
@@ -48,3 +50,9 @@ then
 else
   ./scripts/installChaincode.sh -b $BRANCH_TAG
 fi
+
+if [[ $TEST_MODE == "-t" ]]; then
+  export UC4_KIND_NODE_IP=$(get_worker_ip)
+  printf "Use the following command to set the node ip:\n"
+  printf "export UC4_KIND_NODE_IP=%s\n" $UC4_KIND_NODE_IP
```
Comment (NikoBergemann) [inline deploy.sh:57]:
```suggestion
  ./scripts/startNetwork/setupLocalNodeIP.sh
```

File: scripts/startNetwork.sh [hunk 3]
```diff
@@ -6,15 +6,19 @@ set -e
...
+  ?) printf 'Invalid flag!' ;;
   esac
 done
 
 # Set environment variables
 source ./scripts/env.sh
 source ./scripts/util.sh
 
+if [[ $TEST_MODE == "-t" ]]; then
+  ./scripts/startNetwork/setupLocalNodeIP.sh
```
Comment (NikoBergemann) [inline scripts/startNetwork.sh:19]:
```suggestion
  ./startNetwork/setupLocalNodeIP.sh
```

are we not already in path ./scripts?

File: deploy.sh (hunk 2 above)
Comment (palkra) [reply deploy.sh:57]: The problem is that the environment variable does not live over the scope of that terminal and sourcing causes some undesired side effects :/

File: scripts/startNetwork.sh (hunk 3 above)
Comment (palkra) [reply scripts/startNetwork.sh:19]: We use absolute paths to avoid path problems. The scripts are initially called from the root folder :)

**Review by matthias-geuchen — APPROVED** [review]

**Pull request merged**